### DISCUSSION TITLE
Service release 4 Grid problems

### DISCUSSION THREAD

**Discussion by tom-b-iodigital** [dotnet/maui#14519] [General]
Before I start filing bug reports, can somebody else confirm that grids are completely and utterly broken in service release 4 (at least on android)? It now seems to be impossible to use "auto" and * together

For example in the standard maui template replace the last label before the button with:

```
<Grid>
      <Grid.ColumnDefinitions>
          <ColumnDefinition Width="auto"></ColumnDefinition>
          <ColumnDefinition Width="*"></ColumnDefinition>
      </Grid.ColumnDefinitions>

      <BoxView Color="Red" WidthRequest="20"></BoxView>
      
      <Label
          Grid.Column="1"
          Text="Welcome to .NET Multi-platform App UI"
          SemanticProperties.HeadingLevel="Level2"
          SemanticProperties.Description="Welcome to dot net Multi platform App U I"
          FontSize="18"
          HorizontalOptions="Center" />
  </Grid>
```

The result will be that neither the boxview nor the label are shown, however if you give the first column a fixed width, everything works:

```
<Grid>
    <Grid.ColumnDefinitions>
        <ColumnDefinition Width="20"></ColumnDefinition>
        <ColumnDefinition Width="*"></ColumnDefinition>
    </Grid.ColumnDefinitions>

    <BoxView Color="Red" WidthRequest="20"></BoxView>
    
    <Label
        Grid.Column="1"
        Text="Welcome to .NET Multi-platform App UI"
        SemanticProperties.HeadingLevel="Level2"
        SemanticProperties.Description="Welcome to dot net Multi platform App U I"
        FontSize="18"
        HorizontalOptions="Center" />
</Grid>
```

IMHO this has made grids useless for anything but the most basic use cases and has broken the layout of the app I'm working on completely.

Is this not supposed to work? Or is this an untested consequence of the bug fixes that have been made on the star-colums/rows of the grid component?

**Comment by tom-b-iodigital** [comment]
I opened an issue ticket: https://github.com/dotnet/maui/issues/14520

I realize how much work goes into a product like Maui, but I really don't understand how such a big issue is allowed to occur in a service release. This further proves that as long as Microsoft doesn't use Maui in their own products and takes its development seriously, we can't reliably use it for our own production apps. This is very dissapointing.

**Comment by edoust** [comment]
I opened a similar ticket for Windows #14537

**Comment by danielftz** [comment]
Big problem...

**Comment by EdwardMcFlurry** [comment]
Funny enough, I’ve been putting in my head now from yesterday a very serious and aggressive email to be sent to Google Play Store support and ask why after I released a new update of my App that the Layout and Controls were all of a sudden rearranged and messed up on Android 😒 whereas it’s Microsoft to blame 😳

I’m glad I didn’t sent that email but this is really bad.

@tom-b-iodigital thanks for picking this up as I almost lost it.

**Reply by baaaaif** [reply]
o_O Sorry, but that's not a bit better than what Microsoft has published here.
Who publishes an app update without looking at it at least once? Hopefully we're talking about the beta channel here ;-)
